### ISSUE TITLE
Cannot delete pipeline: cannot_invalidate_during_initialization

### ISSUE THREAD

**Issue by JohannesRudolph** [concourse/concourse#1823]
# Bug Report

Another Concourse babysitting story from today: we added a new resource intensive pipeline to our cluster which caused some of the workers to go out of memory during build, resulting in lots of errored resources and tasks.

We tried to delete the pipeline using `fly destroy-pipeline` to be greeted by
``` 
error: Unexpected Response
Status: 500 Internal Server Error
Body:
```

I could extract this from our ATC logs: 
```
{"timestamp":"1511192344.895966053","source":"atc","message":"atc.destroying-pipeline.start","log_level":1,"data":{"name":"meshfed-gradle","session":"15010"}}
{"timestamp":"1511192344.958673477","source":"atc","message":"atc.destroying-pipeline.failed","log_level":2,"data":{"error":"pq: new row for relation \"volumes\" violates check constraint \"cannot_invalidate_during_initialization\"","name":"meshfed-gradle","session":"15010"}}
```

The pipeline is our first one to use task `caches`. I guess that may be somehow related?
No (visible) builds were running in the pipeline when I tried to delete it. Retrying several times did not change the result. 

The following can also be handy:

* Concourse version: 3.6
* Deployment type: Docker
* Infrastructure/IaaS: Meshcloud OpenStack

**Comment by shashwathi** [comment]
@JohannesRudolph  : This is fixed in concourse 3.9.2 version. Can you verify if you are still seeing this issue?

**Comment by vito** [comment]
This is fixed by #2031.
